### PR TITLE
[WIP] Add Docker Testing to travisCI

### PR DESCRIPTION
**Pull request by canyon289**
Trying to get testing to be done in containers

### CODE REVIEW

File: scripts/Dockerfile [hunk 1]
```diff
@@ -12,7 +12,21 @@ ENV PYTHON_VERSION=${PYTHON_VERSION}
...
-RUN apt-get update && apt-get install -y git build-essential \
+# For Sphinx documentation builds
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+ENV TRAVIS_BRANCH=${TRAVIS_BRANCH}
+ENV TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+ENV TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+ENV TRAVIS_TAG=${TRAVIS_TAG}
+ENV GH_TOKEN=${GH_TOKEN}
+ENV GH_REPO_SLUG=${GH_REPO_SLUG}
```
Comment (ahartikainen) [inline scripts/Dockerfile]: we need at least TRAVIS_JOB_ID

https://docs.travis-ci.com/user/coveralls/#using-coveralls-with-docker-builds

File: scripts/Dockerfile [hunk 2]
```diff
@@ -15,6 +15,16 @@ ENV DOCKER_BUILD=true
...
 ENV LANG=C.UTF-8
+ENV TRAVIS_BRANCH=${TRAVIS_BRANCH}
+ENV TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+ENV TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+ENV TRAVIS_TAG=${TRAVIS_TAG}
+ENV GH_TOKEN=${GH_TOKEN}
+ENV GH_REPO_SLUG=${GH_REPO_SLUG}
+
+# For Coveralls
+ENV COVERALLS_REPO_TOKEN=${COVERALLS_REPO_TOKEN}
```
[the remark this answers is not in the document]
Comment (canyon289) [reply scripts/Dockerfile]: Is the concern someone could write code that would print the token?

File: scripts/Dockerfile (hunk 1 above)
Comment (canyon289) [reply scripts/Dockerfile]: This link is very helpful!

**Comment by canyon289** [comment]
coveralls is working now! The source code is parsed and sent to the coveralls website.

The one thing that is broken is the coveralls drilldown view
https://coveralls.io/jobs/43686645

Here's an example of a working one.
https://coveralls.io/jobs/43682680

We can make the choice of merging this now or fixing coveralls drilldown and merging later.

**Comment by ahartikainen** [comment]
LGTM

**Comment by ahartikainen** [comment]
Hi, I did one test before merge. We can remove it if it doesn't work

**Comment by ahartikainen** [comment]
It did not, se let's rebase and remove that commit and try it in another PR

**Pull request merged**